tests/conditional_compilation/conftest.py

```diff
@@ -71,14 +70,18 @@ def pytest_generate_tests(metafunc):
         test_cases = yaml.safe_load(file)
 
     for test in test_cases:
-        extra_args = {}
-        model_path = test["model"]["path"]
-        if "marks" in test:
-            extra_args["marks"] = test["marks"]
-
-        test_id = model_path.replace("$", "").replace("{", "").replace("}", "")
-        params.append(pytest.param(test_id, Path(expand_env_vars(model_path)), **extra_args))
-        ids = ids + [test_id]
+        model_list = []
+        test_id_list = []
+        for models in test["collection"]:
+            extra_args = {}
+            model_path = models["model"]["path"]
+            if "marks" in test["collection"]:
+                extra_args["marks"] = test["marks"]
+            model_list.append(expand_env_vars(model_path))
+            test_id_list.append(model_path.split("/")[len(model_path.split("/")) - 1].replace(".xml", ""))
```

Comment:
More suitable using following construction `model_path.split("/")[-1]`. This is exclude double call of split method.
And why are you remove file extension?